File: threexp1.nim
proc threexp1(x: int) =
    var x: int = x
    while x != 1:
        if x mod 2 == 0:
            x = x div 2
        else:
            x = 3 * x + 1
        echo x
    echo "OK"

threexp1(27)

File: threexp1.rs
fn threexp1(mut x: isize) {
    while x != 1 {
        if x % 2 == 0 {
            x = x / 2;
            println!("{0}", x);
        } else {
            x = 3 * x + 1;
            println!("{0}", x);
        }
    }
    println!("OK");
}

fn main() {
    threexp1(27);
}
